domain-management/src/main/java/org/jboss/as/domain/management/security/SecurityRealmService.java

```diff
@@ -97,6 +97,8 @@
  */
 public class SecurityRealmService implements Service<SecurityRealm>, SecurityRealm {
 
+    private static final Provider ELYTRON_PROVIDER = new WildFlyElytronProvider();
```

Comment:
As long as it is not created in both providers directly it is fine.
I was testing locally by just having 
`Provider[] providers = new Provider[]{new WildFlyElytronProvider()}`
in service code and than returning that in both providers.